### PR TITLE
TPROD-339 - SocialBundle - Convert PHP Templates to Twig Templates

### PR DESCRIPTION
**Pull request by JoshuaEstes**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. 

- `/s/monitoring`
  - Does page render with and without data
  - Can be filtered
  - Columns sorted correctly
- `/s/monitoring/new`
  - Form/page render correctly
  - Select each **Monitoring Method** to ensure it updates the form correctly
- `/s/monitoring/view/{id}`
  - Does details page render correctly
- `/s/monitoring/edit/{id}`
  - Form/page renders correctly
- `/s/tweets/new`
  - Form/page renders correctly
- `/s/tweets/edit/{id}`
  - Form/page render correctly
- Within the Camping Builder, add the action **Tweet Contact**, Click the **New Tweet** button. Verify the form renders correctly
  - **NOTE**: There are bugs present on the 5.x branch. Specifically, if you save a tweet it doesn't close the window and add it to the list. Instead it just keeps the window open and redirects to the list page. To see the newly created tweet, you need cancel the new action and add the tweet action again.
- Navigate to to `/s/messages/new`, under the **Tweet** tab, select **Enabled?** and for **Message** select the **Create New** option.
  - Verify the modal renders correctly
- `/s/contacts/view/{id}` **requires app be setup for each social platform**
  - Link the contact to a social account using the integration

**Notes**:
- The social login template is covered by the PR https://github.com/mautic/mautic/pull/11898

**REQUIREMENTS**
- Testing this will require an app be setup for: Facebook, Foursquare, Instagram, LinkedIn, and Twitter. This is only for the contact's profile. The rest of the time, you can use dummy keys.

**Docs**
- https://docs.mautic.org/en/channels/social-monitoring
- https://docs.mautic.org/en/contacts/manage-contacts/contact-overview#social

### CODE REVIEW

**Comment by FabienneAivie** [comment]
The first steps worked well. 

`/s/tweets/edit/{id}`

The tweets would not open when clicking on them to edit a tweet. 

:s:tweets:edit:{id} - did not work

**Review by FabienneAivie — CHANGES_REQUESTED** [review]
Opening tweet details/edit does not work.

**Comment by mabumusa1** [comment]
- [X]  Dashboard works
- [X] Search works
- [ ] Monitor Dashboard, error in the title of the page
image
- [X] New Monitoring works
- [X] Edit works
- [X] Categories work 
- [X] Delete work
- [X] Create Tweet
- [X] Edit Tweet
- [X] Delete tweet - there is a bug missing translation
image
- [ ] In tweet dashboard, there is a link to `s/tweets/view/{id}`, this route does not exist 
- [ ] Publish/Unpublish toggle does not work well, it stops after the first click 
- [X] Marketing messages works
- [X] Campaign modals work

**Comment by JoshuaEstes** [comment]
@mabumusa1 

>  Monitor Dashboard, error in the title of the page

I cannot reproduce, can you give me more details on how to get this error?

>  Delete tweet - there is a bug missing translation

Resolved https://github.com/mautic/mautic/pull/11944/commits/ba302de2d9bbc7c2d0bc456c7d02d734a9680ca0

> In tweet dashboard, there is a link to `s/tweets/view/{id}`, this route does not exist

This does not exist on the 5.x branch either. I'm not sure why it acts differently. I made an update to make sure requests are forwarded to the correct route when the view route is hit. https://github.com/mautic/mautic/pull/11944/commits/57a802e868a51ea48c726977b994d63e772ab552

>  Publish/Unpublish toggle does not work well, it stops after the first click

Same behavior on the 5.x branch. I took a look at the bug and can't seem to figure out where the issue is. It looks like we can invest more time tracking the bug down or rewrite the controller. Not sure which is going to be less time.

**Review by RCheesley — APPROVED** [review]
I have tested this with Twitter but I can't get the command to run to actually bring in the tweets to check the layout on the contact profile etc so this is just based on what I have been able to test so far.

Some things I noticed:

- Tab title seems to be OK for me on the social monitoring dashboard - I was not able to reproduce what @mabumusa1 found with that

- Creating a tweet results in a timeout: 

`[2023-02-14 18:51:15] php.CRITICAL: Fatal Error: Allowed memory size of 1073741824 bytes exhausted (tried to allocate 20480 bytes) {"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\OutOfMemoryError(code: 0): Error: Allowed memory size of 1073741824 bytes exhausted (tried to allocate 20480 bytes) at /var/www/html/vendor/symfony/form/Extension/DataCollector/FormDataExtractor.php:44)`
 
This happened from within the campaign builder and also in Channels > Tweets > New - I think this is the bug that @JoshuaEstes mentions.

- Marketing messages works as expected

- All views are working as I would expect - I tried creating mention and hashtag monitors, editing them and deleting them

- I think the toggle thing is a separate and existing issue https://github.com/mautic/mautic/issues/10932

Great job @JoshuaEstes 🚀

**Review by RCheesley — COMMENTED** [review]
There deffo seems to be an issue with the mautic:social:monitoring command.

It's present in the 4.4 branch but missing in the 5.x branch.

I will raise the issue but without proper testing of the actual ingress of data from Twitter I feel we should not merge this PR.

**Comment by RCheesley** [comment]
https://github.com/mautic/mautic/issues/11975 here is the issue I have raised separately

**Comment by JoshuaEstes** [comment]
@RCheesley The SocialBundle covers: Facebook, Foursquare, Instagram, LinkedIn, and Twitter

**Comment by RCheesley** [comment]
Yes but many of them don't work more recently with the API changes - Twitter is the one that is most used in the community as far as I know.

File: plugins/MauticSocialBundle/Translations/en_US/messages.ini [hunk 1]
```diff
@@ -100,7 +100,7 @@ mautic.mauticSocial.permissions.header="Social Permissions"
 mautic.mauticSocial.permissions.monitoring="Monitoring - User has access to"
 mautic.mauticSocial.permissions.tweet="Tweet - User has access to"
 mautic.mautic.social.monitoring.form.confirmdelete="Delete monitor?"
-mautic.mautic.integration.Twitter.form.confirmdelete="Delete tweet?"
+mautic.integration.Twitter.form.confirmdelete="Delete tweet?"
```
Comment (escopecz) [inline plugins/MauticSocialBundle/Translations/en_US/messages.ini:103]: Why was this changed? Now all the language translations will have to translate this again.

File: plugins/MauticSocialBundle/Views/Tweet/form.html.twig [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{#
+#}
+{#% form_theme form with [
+] %#}
```
Comment (escopecz) [inline plugins/MauticSocialBundle/Views/Tweet/form.html.twig]: Is this prepared for something?

**Comment by RCheesley** [comment]
@JoshuaEstes some code review comments for you to check, I have just merged the fix for the social monitoring commands so that we can test this fully :)

**Review by RCheesley — DISMISSED** [review]
Confirmed everything is as expected with the integration enabled - social platform is showing in the right place, the tabs work and all is good with the world! 🎉 🚀 LGTM!

**Comment by RCheesley** [comment]
@JoshuaEstes could you just address the feedback first?

File: plugins/MauticSocialBundle/Translations/en_US/messages.ini (hunk 1 above)
Comment (JoshuaEstes) [reply plugins/MauticSocialBundle/Translations/en_US/messages.ini:103]: https://github.com/mautic/mautic/pull/11944#issuecomment-1426785218

This was the only way I was able to find to get the modal dialog to display. I could be missing something, in which case please let me know.

Comment (escopecz) [reply plugins/MauticSocialBundle/Translations/en_US/messages.ini:103]: You are right. It must heave been broken from the beginning. Thanks for the info!

**Review by escopecz — APPROVED** [review]
Thank you! I see no further issues in the code 👍
